sources/visual/Trajectory.h

```diff
@@ -0,0 +1,24 @@
+#ifndef TRAJECTORY_H
+#define TRAJECTORY_H
+
+#include <vector>
+#include <memory>
+#include <iostream>
+#include "theglobals.h"
+#include <cmath>
+
+// abstract class as a base for every kind of trajectory
+class Trajectory {
+public:
+    virtual std::vector<double> computeTrajectory(double t, double initX, double initY) = 0;
+    Trajectory(double s);
```

Comment:
un petit `explicit` ici, afin d'interdire au compilateur de se permettre une conversion implicite.

```suggestion
    explicit Trajectory(double s);
```